src/plugins/data/public/ui/typeahead/suggestions_component.tsx

```diff
@@ -29,12 +29,44 @@ interface Props {
   show: boolean;
   suggestions: QuerySuggestion[];
   loadMore: () => void;
+  queryBarRect?: DOMRect;
+  dropdownHeight?: string;
 }
 
 export class SuggestionsComponent extends Component<Props> {
   private childNodes: HTMLDivElement[] = [];
   private parentNode: HTMLDivElement | null = null;
 
+  kbnTypeaheadDivRefInstance: RefObject<HTMLDivElement> = createRef();
+
+  updatePosition = () => {
+    const kbnTypeaheadDiv = this.kbnTypeaheadDivRefInstance.current;
+    const { queryBarRect } = this.props;
+
+    if (queryBarRect && kbnTypeaheadDiv) {
+      const documentHeight = document.documentElement.clientHeight || window.innerHeight;
+      const suggestionsListOffsetBottom = 20;
+      const suggestionsListHeight = kbnTypeaheadDiv.clientHeight;
+
+      const isSuggestionsListFittable =
```

Comment:
Could you please provide comment here about logic?

Reply:
Sure, left a comment at this row